refactor(frontend-study-gochicus): js was converted into ts

gochicus opened the pull request:
affects: @lessons/how-to-js, @lessons/how-to-use-functions, @lessons/how-to-use-objects,
@lessons/loops-and-conditions, @practice/actions-with-page, @practice/events-in-page,
@practice/food-delivery

ISSUES CLOSED: #24

gochicus committed:
refactor: js was converted into ts

affects: @lessons/how-to-js, @lessons/how-to-use-functions, @lessons/how-to-use-objects,
@lessons/loops-and-conditions, @practice/actions-with-page, @practice/events-in-page,
@practice/food-delivery

ISSUES CLOSED: #24

ad21ewda1231dsa123das12 reviewed (changes requested):
Что со скоупом?

gochicus commented:
> Что со скоупом?

Весь проект указан в скоупе,  изменения затрагивают все пакеты.  Разве не так нужно?

TorinAsakura commented on lessons/loops-and-conditions/README.md:
```diff
@@ -1,5 +1,5 @@
 #  Описание
-В добавленном в разметку файле `index.js` посредством функции `prompt` в цикле,
+В добавленном в разметку файле `index.ts` посредством функции `prompt` в цикле,
 задаётся два вопроса(за искл. первого вопроса). 
```
задаётся два вопроса…за исключением первого вопроса…что за бред?

themetronome commented:
> > Что со скоупом?
> 
> Весь проект указан в скоупе, изменения затрагивают все пакеты. Разве не так нужно?

https://github.com/Atlantis-Academy/frontend-study-ikorol/pull/26 - посмотри этот пуллреквест, здесь Ирина допустила такую же ошибку

TorinAsakura commented on lessons/how-to-use-objects/src/index.ts:
```diff
@@ -1,18 +1,14 @@
 const personalMoviesDB = {
-  moviesCount: () => {
+  moviesCount: (): void => {
```
почему везде протыкан void? у вас функции вообще ничего не возвращают? Совсем ничего? Хорошо подумали?

TorinAsakura commented on practice/actions-with-page/src/index.ts:
```diff
@@ -0,0 +1,41 @@
+    "Скотт Пилигрим против...",
+  ],
+};
+function replaceOrder(): void {
```
опять 25

TorinAsakura commented on practice/events-on-page/src/index.ts:
```diff
@@ -9,24 +9,24 @@ document.addEventListener("DOMContentLoaded", () => {
 
-  form.addEventListener("submit", (evt) => {
+  form.addEventListener("submit", (evt): void => {
     evt.preventDefault();
```
что такое evt?

TorinAsakura commented on practice/events-on-page/src/index.ts:
```diff
@@ -9,24 +9,24 @@ document.addEventListener("DOMContentLoaded", () => {
+  const checkbox: HTMLInputElement = form.querySelector("[type='checkbox']");
+  const bin: NodeListOf<Element> = document.querySelectorAll(".delete");
 
   bin.forEach((item, i) => {
```
что такое i?

gochicus replied:
@TorinAsakura Это просто вопрос на понимание или мне нужно переименовать аргумент, отвечающий за номер обрабатываемого  элемента массива bin?

TorinAsakura commented on practice/events-on-page/src/index.ts:
```diff
@@ -54,7 +54,7 @@ document.addEventListener("DOMContentLoaded", () => {
 
-  const sortArr = (arr) => {
+  const sortArr = (arr: Array<string>): void => {
     arr.sort();
```
что такое arr?

TorinAsakura commented on practice/food-delivery/src/index.ts:
```diff
@@ -1,9 +1,13 @@
+    ".tabcontent"
+  );
+  const menuItems: NodeListOf<Element> = document.querySelectorAll(
+    ".tabheader__item"
```
что такое tabheader?

TorinAsakura commented on practice/food-delivery/src/index.ts:
```diff
@@ -34,14 +41,14 @@ document.addEventListener("DOMContentLoaded", () => {
+  const deadline: string = "2021-05-11";
+  function getTimeRemaining(endTime: string) {
+    let now: any = new Date();
     const t = Date.parse(endTime) - now,
```
что такое t?

TorinAsakura commented on practice/food-delivery/src/index.ts:
```diff
@@ -51,22 +58,22 @@ document.addEventListener("DOMContentLoaded", () => {
-    const minutes = timer.querySelector("#minutes");
-    const seconds = timer.querySelector("#seconds");
+  function setTimer(selector: any, endTime: any) {
+    const timer: any = document.querySelector(selector);
+    const days: any = timer.querySelector("#days");
+    const hours: any = timer.querySelector("#hours");
+    const minutes: any = timer.querySelector("#minutes");
+    const seconds: any = timer.querySelector("#seconds");
```
почему any?

gochicus committed:
refactor: new patterns implemented, src in index.html changed to folder dist

affects: @lessons/how-to-js, @lessons/how-to-use-functions, @lessons/how-to-use-objects,
@lessons/loops-and-conditions, @practice/actions-with-page, @practice/events-in-page,
@practice/food-delivery

ISSUES CLOSED: #24

TorinAsakura commented on lessons/how-to-js/src/index.ts:
```diff
@@ -0,0 +1,24 @@
+  genres: [],
+  private: false,
+};
+const a: string = prompt("Последний просмотренный фильм?", "");
+const b: number = +prompt("На сколько оцените его?", "");
+const c: string = prompt("Последний просмотренный фильм?", "");
+const d: number = +prompt("На сколько оцените его?", "");
```
https://www.youtube.com/watch?v=NMUsUiFokr4
на этом пока можно закончить

gochicus committed:
refactor: variables renamed, classes renamed

affects: @lessons/how-to-js, @lessons/how-to-use-functions, @lessons/how-to-use-objects,
@lessons/loops-and-conditions, @practice/actions-with-page, @practice/events-in-page,
@practice/food-delivery

ISSUES CLOSED: #24

ad21ewda1231dsa123das12 commented on lessons/how-to-use-functions/src/index.ts:
```diff
@@ -24,11 +24,19 @@ let personalMoviesDB: Database = {
+      lastFilm !== "" &&
+      lastFilm.length < 50
+    ) {
+      return (personalMoviesDB.movies[lastFilm] = filmRating);
```
здесь точно должен быть return?

ad21ewda1231dsa123das12 commented on lessons/how-to-use-objects/src/index.ts:
```diff
@@ -11,11 +11,17 @@ const personalMoviesDB = {
+        filmRating !== "" &&
+        lastFilm.length < 50
+      ) {
+        return (personalMoviesDB.movies[lastFilm] = filmRating);
```
здесь тоже

gochicus committed:
refactor: return removed from rememberMyfilms functions in packages

affects: @lessons/how-to-use-functions, @lessons/how-to-use-objects

ISSUES CLOSED: #24

gochicus committed:
refactor: packages deleted from  lessons package

ISSUES CLOSED: #24

ad21ewda1231dsa123das12 reviewed (approved):
С точки зрения "уборки" - код стал чище, явные фрагменты антипаттернов были устранены. Единственное что, не могу в полной мере оценить указания типов в тех или иных местах, т.к. нахожусь в данный момент, примерно на том же уровне понимания. На мой взгляд, в некоторых местах, например с указанием типов полей в объекте через интерфейс выглядит overkill

TorinAsakura commented on lessons/loops-and-conditions/README.md:
```diff
@@ -1,6 +1,5 @@
 #  Описание
-В добавленном в разметку файле `index.js` посредством функции `prompt` в цикле,
-задаётся два вопроса(за искл. первого вопроса). 
+ Посредством функции `prompt` в цикле,задаётся два вопроса(за искл. первого вопроса). 
```
Сделай нормальное описание уже
